test/test_optim.py

```diff
@@ -1139,6 +1139,21 @@ def test_no_cyclic_references(self):
         self.assertEqual(
             gc.collect(), 0, msg="Optimizer should be garbage-collected on __del__")
 
+    def test_no_cyclic_references_in_step(self):
+        def run():
+            param = torch.empty(10, requires_grad=True)
+            optim = SGD(params=[param], lr=0.5)
+            scheduler = LambdaLR(optim, lambda epoch: 1.0)
+            param.sum().backward()
+            optim.step()
+            scheduler.step()
+
+        import gc
+        gc.collect()
+        run()
+        garbage = gc.collect()
```

Comment:
You don't want the collect here I think?
THe whole point here is not to create ref cycles right?

Reply:
Yes, if there are any ref cycles created in `run()`, the second `gc.collect()` would return non-zero.

Reply:
My bad, I was still under the idea of using weakref and didn't read the update carefully. Sounds good!